src/main/java/com/facebook/openwifirrm/ucentral/UCentralUtils.java

```diff
@@ -111,34 +111,32 @@ private static void extractIEs(
 			logger.debug("Wifiscan entry does not contain 'ies' field.");
 			return;
 		}
-		JsonArray ies = iesJsonElement.getAsJsonArray();
-		for (JsonElement ie : ies) {
+		JsonArray iesJsonArray = iesJsonElement.getAsJsonArray();
+		InformationElements ies = new InformationElements();
+		for (JsonElement ie : iesJsonArray) {
 			JsonElement typeElement = ie.getAsJsonObject().get("type");
 			if (typeElement == null) {
 				continue;
 			}
 			JsonObject contents =
 				ie.getAsJsonObject().get(IE_CONTENT_FIELD_KEY).getAsJsonObject();
-			try {
-				switch (typeElement.getAsInt()) {
-				case Country.TYPE:
-					entry.country = Country.parse(contents);
-					break;
-				case QbssLoad.TYPE:
-					entry.qbssLoad = QbssLoad.parse(contents);
-					break;
-				case LocalPowerConstraint.TYPE:
-					entry.localPowerConstraint =
-						LocalPowerConstraint.parse(contents);
-					break;
-				case TxPwrInfo.TYPE:
-					entry.txPwrInfo = TxPwrInfo.parse(contents);
-					break;
-				}
-			} catch (NullPointerException e) {
-				logger.debug("Skipping invalid IE.", e);
+			switch (typeElement.getAsInt()) {
```

Comment:
I removed the `NullPointerException` check here, but I'm not sure the best way to proceed... We don't want a single invalid IE to crash the `Modeler` thread, right? But doing a null check for every single field instead of catching an exception (which shouldn't happen often) seems like too much extra code? Thoughts?

Based on this, I may change all the IEs. Currently, mandatory fields are assumed to exist in all the IEs...

Reply:
Also, keep in mind that the plan is to also parse in maybe another 10 IEs...

Reply:
If you really feel it is too much code, you can `catch (Exception e)` here (not specifically `NullPointerException`)

Reply:
I will catch `Exception`